CTPPS: Geometry for UFSDetectors

forthommel opened the pull request:
This PR defines the geometry for the UFSD planes replacing the diamonds' planes 4 in sectors 45 and 56 in 2017.
It also redefines a few diamond volumes' parentage to ease the alignment procedure for all subdetectors.

The re-reco of 2017 data with UFSDs is dependent on this PR.

`runTheMatrix.py -l limited` yielded `19 18 17 14 8 1 1 1 tests passed, 1 1 0 0 0 0 0 0 failed` (usual `4.22` DAS problem, and "Connection timed out problem" while opening `RecoPixelVertexing/PixelLowPtUtilities/data/pixelShapePhase1_noL1.par` in step2 of `7.3`)

vsola committed:
Initial import of the UFSD geometry

forthommel committed:
Updated the geometry testing configuration

nminafra committed:
Updated Geometry for UFSD

mackoo13 committed:
timing detectors geometry (from Nicola Minafra and CTPPS repository)

forthommel committed:
Removed the unused UFSD materials list (to avoid information duplication) ; started to modify the detectors assembly to include the UFSD plane (now, at position 4...)

forthommel committed:
Reusing the diamond detector assembly to include the UFSD plane ; small correction in the definition of the z size of the strip

forthommel committed:
updated DQM test

forthommel committed:
Updated UFSD geometry, dqm test on recent run

forthommel committed:
Integration of the UFSD geometrical elements names in a 9_4_X environment

forthommel committed:
bug fixing

Conflicts:

	DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

forthommel committed:
Reverted back the DQM modifications

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20548/748

cmsbuild commented:
A new Pull Request was created by  @forthommel (Laurent Forthomme) for master.

It involves the following packages:

Geometry/VeryForwardData
Geometry/VeryForwardGeometry
Geometry/VeryForwardGeometryBuilder


@cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please review it and eventually sign? Thanks.
@ghellwig this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

forthommel commented:
Updated the PR description with the result of `runTheMatrix.py` on top of d8323448c833404ce9855eb12696171df2d44e25.

ianna commented:
Please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d8323448c833404ce9855eb12696171df2d44e25
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23071/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/e133687c2dd23771612485bf4baf046b9317e8e8
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23071/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23071/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23071/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 28 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2649068
 - DQMHistoTests: Total failures: 242
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2648637
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Planes/CTPPS_Diamond_Plane1.xml:
```diff
@@ -29,12 +29,12 @@
-      <Translation x="-[CTPPS_Diamond_Parameters:Diamond_dx]*2+[CTPPS_Diamond_Parameters:PCB_dx]/2+[CTPPS_Diamond_Parameters:PCB_extend_dx]" y="0*mm" z="([CTPPS_Diamond_Parameters:Diamond_dz]+[CTPPS_Diamond_Parameters:Metalized_thick]+[CTPPS_Diamond_Parameters:PCB_thick])/2"/>
-      <rRotation name="rotations:000D"/>
-    </PosPart>
+    <!--<PosPart copyNumber="1">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Detector_Assembly.xml:
```diff
@@ -61,12 +61,24 @@
       <Translation y="0*cm" x="[Shift_dx]" z="([CTPPS_Diamond_Parameters:Plane_dz]+[CTPPS_Diamond_Parameters:Plane_gap])/2"/>
       <rRotation name="CTPPS_Diamond_Transformations:planes_x_rotation"/>
     </PosPart>
+    <!--<PosPart copyNumber="4">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Detector_Assembly.xml:
```diff
@@ -61,12 +61,24 @@
       <rParent name="CTPPS_Diamond_Main_Box"/>
+      <rChild name="CTPPS_UFSD_Plane4:CTPPS_UFSD_Plane"/>
+      <Translation y="0*cm" x="[Shift_dx]" z="([CTPPS_Diamond_Parameters:Plane_dz]+[CTPPS_Diamond_Parameters:Plane_gap])*3/2"/>
+<!--       <rRotation name="CTPPS_Diamond_Transformations:planes_x_rotation"/> -->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Detector_Assembly.xml:
```diff
@@ -61,12 +61,24 @@
+      <Translation y="0*cm" x="[Shift_dx]" z="([CTPPS_Diamond_Parameters:Plane_dz]+[CTPPS_Diamond_Parameters:Plane_gap])*3/2"/>
+<!--       <rRotation name="CTPPS_Diamond_Transformations:planes_x_rotation"/> -->
+    </PosPart>
+<!--    <PosPart copyNumber="4">
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Planes/CTPPS_Diamond_Plane2.xml:
```diff
@@ -29,12 +29,12 @@
-      <Translation x="-[CTPPS_Diamond_Parameters:Diamond_dx]*2+[CTPPS_Diamond_Parameters:PCB_dx]/2+[CTPPS_Diamond_Parameters:PCB_extend_dx]" y="0*mm" z="([CTPPS_Diamond_Parameters:Diamond_dz]+[CTPPS_Diamond_Parameters:Metalized_thick]+[CTPPS_Diamond_Parameters:PCB_thick])/2"/>
-      <rRotation name="rotations:000D"/>
-    </PosPart>
+    <!--<PosPart copyNumber="2">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Planes/CTPPS_Diamond_Plane3.xml:
```diff
@@ -29,12 +29,12 @@
-      <Translation x="-[CTPPS_Diamond_Parameters:Diamond_dx]*2+[CTPPS_Diamond_Parameters:PCB_dx]/2+[CTPPS_Diamond_Parameters:PCB_extend_dx]" y="0*mm" z="([CTPPS_Diamond_Parameters:Diamond_dz]+[CTPPS_Diamond_Parameters:Metalized_thick]+[CTPPS_Diamond_Parameters:PCB_thick])/2"/>
-      <rRotation name="rotations:000D"/>
-    </PosPart>
+    <!--<PosPart copyNumber="3">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Planes/CTPPS_Diamond_Plane4.xml:
```diff
@@ -29,12 +29,12 @@
-      <Translation x="-[CTPPS_Diamond_Parameters:Diamond_dx]*2+[CTPPS_Diamond_Parameters:PCB_dx]/2+[CTPPS_Diamond_Parameters:PCB_extend_dx]" y="0*mm" z="([CTPPS_Diamond_Parameters:Diamond_dz]+[CTPPS_Diamond_Parameters:Metalized_thick]+[CTPPS_Diamond_Parameters:PCB_thick])/2"/>
-      <rRotation name="rotations:000D"/>
-    </PosPart>
+    <!--<PosPart copyNumber="4">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Sensitive_Dets.xml:
```diff
@@ -0,0 +1,16 @@
+              xsi:schemaLocation="http://www.cern.ch/cms/DDL ../../DDLSchema/DDLSchema.xsd">
+    <SpecParSection label="CTPPS_Diamond_Sensitive_Dets.xml">
+        <SpecPar name="ROUHitsCTPPSDiamond">
+            <!--  <PartSelector path="//Pat.*_Str.*:Diamond"/>-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Diamond_Sensitive_Dets.xml:
```diff
@@ -0,0 +1,16 @@
+
+            <PartSelector path="//CTPPS_Diamond_Pattern.*_Segment.*:CTPPS_Diamond_Segment"/>
+
+            <!--   <PartSelector path="//PPS_Timing_Box_.*:BoxOFlbar"/>-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_Timing_Positive_Station.xml:
```diff
@@ -41,24 +41,24 @@
-      <Translation x="0*mm" y="0*mm" z="([CTPPS_Timing_Station_Parameters:Station_Length]/2-[CTPPS_Timing_Station_Parameters:Station_Pipe_Length]/2)"/>
+      <Translation x="0*mm" y="0*mm" z="-([CTPPS_Timing_Station_Parameters:Station_Length]/2-([CTPPS_Timing_Station_Parameters:Station_Pipe_Length]+[RP_220_Right_Station:RP_220_Right_Prim_Hor_z]+[RP_Device:RP_Device_Length_z]/2)/2)"/>
     </PosPart>
+    <!--<PosPart copyNumber="1">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/CTPPS_UFSD_Planes/CTPPS_UFSD_Plane4.xml:
```diff
@@ -0,0 +1,272 @@
+
+  <PosPartSection label="CTPPS_UFSD_Plane4.xml">
+
+    <!--<PosPart copyNumber="4">-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/RP_220_Left_Station.xml:
```diff
@@ -112,11 +112,11 @@
-            <rChild name="RP_220_Left_Station_Tube_3"/>
-            <Translation x="0*mm" y="0*mm" z="-(-[RP_220_Left_Station_Length]/2+([RP_220_Left_Prim_Hor_z]+[RP_220_Left_Sec_Hor_z])/2)"/>
-        </PosPart>
+        <!--<PosPart>-->
```
@forthommel - please remove commented out lines

ianna commented on Geometry/VeryForwardData/data/RP_220_Right_Station.xml:
```diff
@@ -112,11 +112,11 @@
-            <rChild name="RP_220_Right_Station_Tube_3"/>
-            <Translation x="0*mm" y="0*mm" z="-[RP_220_Right_Station_Length]/2+([RP_220_Right_Prim_Hor_z]+[RP_220_Right_Sec_Hor_z])/2"/>
-        </PosPart>
+        <!--<PosPart>-->
```
@forthommel - please remove commented out lines

mackoo13 committed:
UFSD Geometry fix (with Laurent and Nicola)

forthommel committed:
Cleanup of UFSD and diamond DDL comments

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20548/841

cmsbuild commented:
Pull request #20548 was updated. @cmsbuild, @civanch, @Dr15Jones, @ianna, @mdhildreth can you please check and sign again.

forthommel commented:
Hello @ianna,
Many thanks for your fast review! I removed this extra set of comments in https://github.com/cms-sw/cmssw/pull/20548/commits/3823603a7cc4fe9c73876307debdfb3b9987a003. In the meantime, we discovered a slight offset in the UFSD plane definition (and simplified the volumes parentage) only affecting our local simulation part (hence, not the reconstruction itself). We introduced this fix in https://github.com/cms-sw/cmssw/pull/20548/commits/81fbf0753d7d287e372ae6c2fe4c10f5ba8269d5.

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 81fbf0753d7d287e372ae6c2fe4c10f5ba8269d5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23109/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5e53c2b9539aa99597276fc2af455b7acffca3cf
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23109/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23109/git-merge-result

cmsbuild commented:
Comparison job queued.

ianna commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20548/23109/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 28 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2649068
 - DQMHistoTests: Total failures: 255
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2648624
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

davidlange6 commented:
+1
